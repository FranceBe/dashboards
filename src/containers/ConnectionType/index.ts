import ConnectionTypeContainer, {
  ConnectionType,
} from 'containers/ConnectionType/ConnectionType.container'

export { ConnectionType }

export default ConnectionTypeContainer
