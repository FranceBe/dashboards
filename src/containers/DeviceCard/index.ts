import DeviceCardComponent, { DeviceCard } from 'containers/DeviceCard/DeviceCard.container'

export { DeviceCard }

export default DeviceCardComponent
