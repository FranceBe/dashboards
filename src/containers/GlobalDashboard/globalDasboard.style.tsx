import styled from 'styled-components'

export const Dashboard = styled.div`
  display: flex;
  justify-content: space-between;
`
export const InfoTableContainer = styled.div`
  width: 50%;
`

export const ChartsContainer = styled.div`
  display: flex;
  flex-direction: column;
`
