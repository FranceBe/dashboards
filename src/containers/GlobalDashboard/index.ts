import GlobalDashboardContainer, {
  GlobalDashboard,
} from 'containers/GlobalDashboard/GlobalDashboard.container'

export { GlobalDashboard }

export default GlobalDashboardContainer
