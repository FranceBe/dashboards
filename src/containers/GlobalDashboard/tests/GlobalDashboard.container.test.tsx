import '@testing-library/jest-dom/extend-expect'

import { render, screen } from '@testing-library/react'
import { GlobalDashboard } from 'containers/GlobalDashboard'
import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { momentFormatter } from 'utils/momentFormatter'
import * as customHook from 'utils/useFetch'

describe('GlobalDashboard', () => {
  // Set up
  // Mock fetch response so we do not need to call real API for testing
  const spy = jest.spyOn(customHook, 'useFetch')
  spy.mockReturnValue({
    data: [
      {
        connection_type: 'ethernet',
        last_seen_at: '2021-09-26T15:10:44.987',
        mac_wifi: 'c5:d7:14:84:f8:cf',
        serial_number: 'device_0',
        sim_id: 98962001,
        status: 'connected',
        url: 'http://url.com',
        voltage: 1234.933884,
      },
      {
        connection_type: 'wifi',
        last_seen_at: '2021-09-26T15:10:43.987',
        mac_wifi: 'c4:d7:14:84:f8:cf',
        serial_number: 'device_1',
        sim_id: 98962002,
        status: 'disconnected',
        url: 'http://url.com',
        voltage: 14.84,
      },
      {
        connection_type: 'cellular',
        last_seen_at: '2021-09-26T15:10:42.987',
        mac_wifi: 'a4:d7:14:84:f8:cf',
        serial_number: 'device_2',
        sim_id: 98962003,
        status: 'connected',
        url: 'http://url.com',
        voltage: 3431.84,
      },
    ],
    status: 'fetched',
  })

  // Setup Test with render
  const setupTest = () => {
    return render(
      <Router>
        <GlobalDashboard />
      </Router>,
    )
  }
  it('should match snapshot', () => {
    // We need to await the render because GlobalDashboard use
    // The useFetch hooks
    const { container } = setupTest()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should display VoltageStat', () => {
    setupTest()

    expect(screen.getByText('voltage')).toBeInTheDocument()
    // We test the presence of ChartBar component used in VoltageStat
    expect(screen.getByTestId('chart-bar')).toBeInTheDocument()
  })

  it('should display ConnectionTypeStat', () => {
    setupTest()

    // We test the presence of ChartPie component used in ConnectionTypeStat
    expect(screen.getByTestId('chart-pie')).toBeInTheDocument()
  })

  it('should display InfoTable & StatusStat', () => {
    setupTest()

    // We test the presence of each serial_number as text
    // We should find 2 times the serial_number as text
    // because it is rendered once in ChartBar (as axis legend)
    expect(screen.getAllByText('device_0')).toHaveLength(2)
    expect(screen.getAllByText('device_1')).toHaveLength(2)
    expect(screen.getAllByText('device_2')).toHaveLength(2)
    // We test the presence of each mac_wifi as text
    expect(screen.getByText('c5:d7:14:84:f8:cf')).toBeInTheDocument()
    expect(screen.getByText('c4:d7:14:84:f8:cf')).toBeInTheDocument()
    expect(screen.getByText('a4:d7:14:84:f8:cf')).toBeInTheDocument()
    // We test the presence of each sim_id as text
    expect(screen.getByText('98962001')).toBeInTheDocument()
    expect(screen.getByText('98962002')).toBeInTheDocument()
    expect(screen.getByText('98962003')).toBeInTheDocument()
    // We test the presence of each last_seen_at as converted date text
    const dateItem1 = momentFormatter('2021-09-26T15:10:44.987')
    const dateItem2 = momentFormatter('2021-09-26T15:10:43.987')
    const dateItem3 = momentFormatter('2021-09-26T15:10:42.987')
    expect(screen.getByText(dateItem1)).toBeInTheDocument()
    expect(screen.getByText(dateItem2)).toBeInTheDocument()
    expect(screen.getByText(dateItem3)).toBeInTheDocument()
    // We test the presence of each connection_type as icon
    expect(screen.getByTestId('icon-cellular')).toBeInTheDocument()
    expect(screen.getByTestId('icon-wifi')).toBeInTheDocument()
    expect(screen.getByTestId('icon-ethernet')).toBeInTheDocument()
    // We test the presence of each status as icon
    // Icons should display twice : once in InfoTable and once in StatusStat
    expect(screen.getAllByTestId('signal-off')).toHaveLength(2)
    expect(screen.getAllByTestId('signal-on')).toHaveLength(3) // signal on is
    // displayed 3 times, once in StatusStat and 2 in InfoTable because there are 2 devices that
    // are connected in the mocked data for this test
  })
})
