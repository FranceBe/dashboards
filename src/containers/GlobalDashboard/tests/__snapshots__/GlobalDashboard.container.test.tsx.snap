// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalDashboard GlobalDashboard should match snapshot 1`] = `
<div
  class="sc-dlnjwi jNugBK"
>
  <div
    class="sc-eCApnc jLKQe"
  >
    <div>
      <div
        data-testid="chart-bar"
      >
        <div
          class="recharts-wrapper"
          style="position: relative; cursor: default; width: 500px; height: 300px;"
        >
          <svg
            class="recharts-surface"
            height="300"
            version="1.1"
            viewBox="0 0 500 300"
            width="500"
          >
            <defs>
              <clippath
                id="recharts1-clip"
              >
                <rect
                  height="260"
                  width="430"
                  x="65"
                  y="5"
                />
              </clippath>
            </defs>
            <g
              class="recharts-cartesian-grid"
            >
              <g
                class="recharts-cartesian-grid-horizontal"
              >
                <line
                  fill="none"
                  height="260"
                  offset="[object Object]"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="430"
                  x="65"
                  x1="65"
                  x2="495"
                  y="5"
                  y1="265"
                  y2="265"
                />
                <line
                  fill="none"
                  height="260"
                  offset="[object Object]"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="430"
                  x="65"
                  x1="65"
                  x2="495"
                  y="5"
                  y1="200"
                  y2="200"
                />
                <line
                  fill="none"
                  height="260"
                  offset="[object Object]"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="430"
                  x="65"
                  x1="65"
                  x2="495"
                  y="5"
                  y1="135"
                  y2="135"
                />
                <line
                  fill="none"
                  height="260"
                  offset="[object Object]"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="430"
                  x="65"
                  x1="65"
                  x2="495"
                  y="5"
                  y1="70"
                  y2="70"
                />
                <line
                  fill="none"
                  height="260"
                  offset="[object Object]"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="430"
                  x="65"
                  x1="65"
                  x2="495"
                  y="5"
                  y1="5"
                  y2="5"
                />
              </g>
              <g
                class="recharts-cartesian-grid-vertical"
              >
                <line
                  fill="none"
                  height="260"
                  offset="[object Object]"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="430"
                  x="65"
                  x1="136.66666666666669"
                  x2="136.66666666666669"
                  y="5"
                  y1="5"
                  y2="265"
                />
                <line
                  fill="none"
                  height="260"
                  offset="[object Object]"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="430"
                  x="65"
                  x1="280"
                  x2="280"
                  y="5"
                  y1="5"
                  y2="265"
                />
                <line
                  fill="none"
                  height="260"
                  offset="[object Object]"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="430"
                  x="65"
                  x1="423.33333333333337"
                  x2="423.33333333333337"
                  y="5"
                  y1="5"
                  y2="265"
                />
                <line
                  fill="none"
                  height="260"
                  offset="[object Object]"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="430"
                  x="65"
                  x1="65"
                  x2="65"
                  y="5"
                  y1="5"
                  y2="265"
                />
                <line
                  fill="none"
                  height="260"
                  offset="[object Object]"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="430"
                  x="65"
                  x1="495"
                  x2="495"
                  y="5"
                  y1="5"
                  y2="265"
                />
              </g>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
            >
              <line
                class="recharts-cartesian-axis-line"
                fill="none"
                height="30"
                orientation="bottom"
                stroke="#666"
                type="category"
                width="430"
                x="65"
                x1="65"
                x2="495"
                y="265"
                y1="265"
                y2="265"
              />
              <g
                class="recharts-cartesian-axis-ticks"
              >
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="30"
                    orientation="bottom"
                    stroke="#666"
                    type="category"
                    width="430"
                    x="65"
                    x1="136.66666666666669"
                    x2="136.66666666666669"
                    y="265"
                    y1="271"
                    y2="265"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height="30"
                    orientation="bottom"
                    stroke="none"
                    text-anchor="middle"
                    type="category"
                    width="430"
                    x="136.66666666666669"
                    y="273"
                  >
                    <tspan
                      dy="0.71em"
                      x="136.66666666666669"
                    >
                      device_0
                    </tspan>
                  </text>
                </g>
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="30"
                    orientation="bottom"
                    stroke="#666"
                    type="category"
                    width="430"
                    x="65"
                    x1="280"
                    x2="280"
                    y="265"
                    y1="271"
                    y2="265"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height="30"
                    orientation="bottom"
                    stroke="none"
                    text-anchor="middle"
                    type="category"
                    width="430"
                    x="280"
                    y="273"
                  >
                    <tspan
                      dy="0.71em"
                      x="280"
                    >
                      device_1
                    </tspan>
                  </text>
                </g>
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="30"
                    orientation="bottom"
                    stroke="#666"
                    type="category"
                    width="430"
                    x="65"
                    x1="423.33333333333337"
                    x2="423.33333333333337"
                    y="265"
                    y1="271"
                    y2="265"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height="30"
                    orientation="bottom"
                    stroke="none"
                    text-anchor="middle"
                    type="category"
                    width="430"
                    x="423.33333333333337"
                    y="273"
                  >
                    <tspan
                      dy="0.71em"
                      x="423.33333333333337"
                    >
                      device_2
                    </tspan>
                  </text>
                </g>
              </g>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
            >
              <line
                class="recharts-cartesian-axis-line"
                fill="none"
                height="260"
                orientation="left"
                stroke="#666"
                type="number"
                width="60"
                x="5"
                x1="65"
                x2="65"
                y="5"
                y1="5"
                y2="265"
              />
              <g
                class="recharts-cartesian-axis-ticks"
              >
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="260"
                    orientation="left"
                    stroke="#666"
                    type="number"
                    width="60"
                    x="5"
                    x1="59"
                    x2="65"
                    y="5"
                    y1="265"
                    y2="265"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height="260"
                    orientation="left"
                    stroke="none"
                    text-anchor="end"
                    type="number"
                    width="60"
                    x="57"
                    y="265"
                  >
                    <tspan
                      dy="0.355em"
                      x="57"
                    >
                      0
                    </tspan>
                  </text>
                </g>
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="260"
                    orientation="left"
                    stroke="#666"
                    type="number"
                    width="60"
                    x="5"
                    x1="59"
                    x2="65"
                    y="5"
                    y1="200"
                    y2="200"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height="260"
                    orientation="left"
                    stroke="none"
                    text-anchor="end"
                    type="number"
                    width="60"
                    x="57"
                    y="200"
                  >
                    <tspan
                      dy="0.355em"
                      x="57"
                    >
                      900
                    </tspan>
                  </text>
                </g>
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="260"
                    orientation="left"
                    stroke="#666"
                    type="number"
                    width="60"
                    x="5"
                    x1="59"
                    x2="65"
                    y="5"
                    y1="135"
                    y2="135"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height="260"
                    orientation="left"
                    stroke="none"
                    text-anchor="end"
                    type="number"
                    width="60"
                    x="57"
                    y="135"
                  >
                    <tspan
                      dy="0.355em"
                      x="57"
                    >
                      1800
                    </tspan>
                  </text>
                </g>
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="260"
                    orientation="left"
                    stroke="#666"
                    type="number"
                    width="60"
                    x="5"
                    x1="59"
                    x2="65"
                    y="5"
                    y1="70"
                    y2="70"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height="260"
                    orientation="left"
                    stroke="none"
                    text-anchor="end"
                    type="number"
                    width="60"
                    x="57"
                    y="70"
                  >
                    <tspan
                      dy="0.355em"
                      x="57"
                    >
                      2700
                    </tspan>
                  </text>
                </g>
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="260"
                    orientation="left"
                    stroke="#666"
                    type="number"
                    width="60"
                    x="5"
                    x1="59"
                    x2="65"
                    y="5"
                    y1="5"
                    y2="5"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height="260"
                    orientation="left"
                    stroke="none"
                    text-anchor="end"
                    type="number"
                    width="60"
                    x="57"
                    y="5"
                  >
                    <tspan
                      dy="0.355em"
                      x="57"
                    >
                      3600
                    </tspan>
                  </text>
                </g>
              </g>
            </g>
            <g
              class="recharts-layer recharts-bar"
            >
              <g
                class="recharts-layer recharts-bar-rectangles"
              >
                <g
                  class="recharts-layer"
                >
                  <g
                    class="recharts-layer recharts-bar-rectangle"
                  />
                  <g
                    class="recharts-layer recharts-bar-rectangle"
                  />
                  <g
                    class="recharts-layer recharts-bar-rectangle"
                  />
                </g>
              </g>
            </g>
          </svg>
          <div
            class="recharts-legend-wrapper"
            style="position: absolute; width: 490px; height: 320px; left: 5px; bottom: 5px;"
          >
            <ul
              class="recharts-default-legend"
              style="padding: 0px; margin: 0px; text-align: center;"
            >
              <li
                class="recharts-legend-item legend-item-0"
                style="display: inline-block; margin-right: 10px;"
              >
                <svg
                  class="recharts-surface"
                  height="14"
                  style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                  version="1.1"
                  viewBox="0 0 32 32"
                  width="14"
                >
                  <path
                    class="recharts-legend-icon"
                    d="M0,4h32v24h-32z"
                    fill="#0E074B"
                    stroke="none"
                  />
                </svg>
                <span
                  class="recharts-legend-item-text"
                  style="color: rgb(14, 7, 75);"
                >
                  voltage
                </span>
              </li>
            </ul>
          </div>
          <div
            class="recharts-tooltip-wrapper"
            style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px; left: 0px;"
          >
            <div
              class="recharts-default-tooltip"
              style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"
            >
              <p
                class="recharts-tooltip-label"
                style="margin: 0px;"
              >
                0
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-testid="chart-pie"
    >
      <div
        class="recharts-wrapper"
        style="position: relative; cursor: default; width: 400px; height: 400px;"
      >
        <svg
          class="recharts-surface"
          cx="50%"
          cy="50%"
          height="400"
          version="1.1"
          viewBox="0 0 400 400"
          width="400"
        >
          <defs>
            <clippath
              id="recharts4-clip"
            >
              <rect
                height="390"
                width="390"
                x="5"
                y="5"
              />
            </clippath>
          </defs>
          <g
            class="recharts-layer recharts-pie"
          >
            <g
              class="recharts-layer"
            >
              <g
                class="recharts-layer recharts-pie-sector"
              />
              <g
                class="recharts-layer recharts-pie-sector"
              />
              <g
                class="recharts-layer recharts-pie-sector"
              />
            </g>
          </g>
        </svg>
        <div
          class="recharts-tooltip-wrapper"
          style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px; left: 0px;"
        >
          <div
            class="sc-gtsrHT dZVimF"
          >
            ethernet
            : 
            0
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="sc-hKFxyN gcWuxB"
  >
    <div
      class="sc-iJCRrE fqqaic"
    >
      <div
        class="sc-ezzafa iHcNzY"
      >
        <div
          class="sc-iCoGMd bGEYcm rdt_Table"
          role="table"
        >
          <div
            class="sc-fujyAs fjYwFu rdt_TableHead"
            role="rowgroup"
          >
            <div
              class="sc-pNWdM lfjlYU rdt_TableHeadRow"
              role="row"
            >
              <div
                class="sc-jrsJWt sc-kEqXSa sc-fFSPTT hCPiRP dlLhQd cZNhtN rdt_TableCol"
                data-column-id="1"
              >
                <div
                  class="sc-bkbkJK nDHnj rdt_TableCol_Sortable"
                  data-column-id="1"
                  data-sort-id="1"
                  role="columnheader"
                  tabindex="0"
                >
                  <div
                    class="sc-iemWCZ ifvWin"
                    data-column-id="1"
                  >
                    serial_number
                  </div>
                  <span
                    class="sc-fnVZcZ hAsdXb"
                  >
                    ▲
                  </span>
                </div>
              </div>
              <div
                class="sc-jrsJWt sc-kEqXSa sc-fFSPTT hCPiRP dlLhQd cZNhtN rdt_TableCol"
                data-column-id="2"
              >
                <div
                  class="sc-bkbkJK nDHnj rdt_TableCol_Sortable"
                  data-column-id="2"
                  data-sort-id="2"
                  role="columnheader"
                  tabindex="0"
                >
                  <div
                    class="sc-iemWCZ ifvWin"
                    data-column-id="2"
                  >
                    mac_wifi
                  </div>
                  <span
                    class="sc-fnVZcZ hAsdXb"
                  >
                    ▲
                  </span>
                </div>
              </div>
              <div
                class="sc-jrsJWt sc-kEqXSa sc-fFSPTT hCPiRP dlLhQd cZNhtN rdt_TableCol"
                data-column-id="3"
              >
                <div
                  class="sc-bkbkJK nDHnj rdt_TableCol_Sortable"
                  data-column-id="3"
                  data-sort-id="3"
                  role="columnheader"
                  tabindex="0"
                >
                  <div
                    class="sc-iemWCZ ifvWin"
                    data-column-id="3"
                  >
                    sim_id
                  </div>
                  <span
                    class="sc-fnVZcZ hAsdXb"
                  >
                    ▲
                  </span>
                </div>
              </div>
              <div
                class="sc-jrsJWt sc-kEqXSa sc-fFSPTT hCPiRP dlLhQd cZNhtN rdt_TableCol"
                data-column-id="4"
              >
                <div
                  class="sc-bkbkJK nDHnj rdt_TableCol_Sortable"
                  data-column-id="4"
                  data-sort-id="4"
                  role="columnheader"
                  tabindex="0"
                >
                  <div
                    class="sc-iemWCZ ifvWin"
                    data-column-id="4"
                  >
                    last_seen_at
                  </div>
                  <span
                    class="sc-fnVZcZ hAsdXb"
                  >
                    ▲
                  </span>
                </div>
              </div>
              <div
                class="sc-jrsJWt sc-kEqXSa sc-fFSPTT hCPiRP spDng cZNhtN rdt_TableCol"
                data-column-id="5"
              >
                <div
                  class="sc-bkbkJK nDHnj rdt_TableCol_Sortable"
                  data-column-id="5"
                  data-sort-id="5"
                  role="columnheader"
                  tabindex="0"
                >
                  <div
                    class="sc-iemWCZ ifvWin"
                    data-column-id="5"
                  >
                    connection_type
                  </div>
                  <span
                    class="sc-fnVZcZ hAsdXb"
                  >
                    ▲
                  </span>
                </div>
              </div>
              <div
                class="sc-jrsJWt sc-kEqXSa sc-fFSPTT hCPiRP spDng cZNhtN rdt_TableCol"
                data-column-id="6"
              >
                <div
                  class="sc-bkbkJK nDHnj rdt_TableCol_Sortable"
                  data-column-id="6"
                  data-sort-id="6"
                  role="columnheader"
                  tabindex="0"
                >
                  <div
                    class="sc-iemWCZ ifvWin"
                    data-column-id="6"
                  >
                    status
                  </div>
                  <span
                    class="sc-fnVZcZ hAsdXb"
                  >
                    ▲
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="sc-lmgQwP kppOq rdt_TableBody"
            role="rowgroup"
          >
            <div
              class="sc-hBMUJo jzmWhL rdt_TableRow"
              id="row-0"
              role="row"
            >
              <div
                class="sc-jrsJWt sc-kEqXSa sc-iqAclL eAcHFi dlLhQd eZwRKE rdt_TableCell"
                data-column-id="1"
                data-tag="allowRowEvents"
                id="cell-1-undefined"
                role="gridcell"
              >
                <div
                  data-tag="allowRowEvents"
                >
                  <div
                    class="device_0"
                  >
                    device_0
                  </div>
                </div>
              </div>
              <div
                class="sc-jrsJWt sc-kEqXSa sc-iqAclL eAcHFi dlLhQd eZwRKE rdt_TableCell"
                data-column-id="2"
                data-tag="allowRowEvents"
                id="cell-2-undefined"
                role="gridcell"
              >
                <div
                  data-tag="allowRowEvents"
                >
                  c5:d7:14:84:f8:cf
                </div>
              </div>
              <div
                class="sc-jrsJWt sc-kEqXSa sc-iqAclL eAcHFi dlLhQd eZwRKE rdt_TableCell"
                data-column-id="3"
                data-tag="allowRowEvents"
                id="cell-3-undefined"
                role="gridcell"
              >
                <div
                  data-tag="allowRowEvents"
                >
                  98962001
                </div>
              </div>
              <div
                class="sc-jrsJWt sc-kEqXSa sc-iqAclL eAcHFi dlLhQd eZwRKE rdt_TableCell"
                data-column-id="4"
                data-tag="allowRowEvents"
                id="cell-4-undefined"
                role="gridcell"
              >
                <div
                  data-tag="allowRowEvents"
                >
                  September 26th 2021, 3:10:44 pm
                </div>
              </div>
              <div
                class="sc-jrsJWt sc-kEqXSa sc-iqAclL eAcHFi spDng eZwRKE rdt_TableCell"
                data-column-id="5"
                data-tag="allowRowEvents"
                id="cell-5-undefined"
                role="gridcell"
              >
                <div
                  data-tag="allowRowEvents"
                >
                  <div
                    class="sc-jSFjdj gpMpen"
                    data-testid="connection_type-container"
                    type="ethernet"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      data-testid="icon-ethernet"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7.77 6.76L6.23 5.48.82 12l5.41 6.52 1.54-1.28L3.42 12l4.35-5.24zM7 13h2v-2H7v2zm10-2h-2v2h2v-2zm-6 2h2v-2h-2v2zm6.77-7.52l-1.54 1.28L20.58 12l-4.35 5.24 1.54 1.28L23.18 12l-5.41-6.52z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div
                class="sc-jrsJWt sc-kEqXSa sc-iqAclL eAcHFi spDng eZwRKE rdt_TableCell"
                data-column-id="6"
                data-tag="allowRowEvents"
                id="cell-6-undefined"
                role="gridcell"
              >
                <div
                  data-tag="allowRowEvents"
                >
                  <div
                    class="sc-gKAaRy cMtfuc"
                    data-testid="status-container"
                    status="connected"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      data-testid="signal-on"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M2 22h20V2z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="sc-hBMUJo jzmWhL rdt_TableRow"
              id="row-1"
              role="row"
            >
              <div
                class="sc-jrsJWt sc-kEqXSa sc-iqAclL eAcHFi dlLhQd eZwRKE rdt_TableCell"
                data-column-id="1"
                data-tag="allowRowEvents"
                id="cell-1-undefined"
                role="gridcell"
              >
                <div
                  data-tag="allowRowEvents"
                >
                  <div
                    class="device_1"
                  >
                    device_1
                  </div>
                </div>
              </div>
              <div
                class="sc-jrsJWt sc-kEqXSa sc-iqAclL eAcHFi dlLhQd eZwRKE rdt_TableCell"
                data-column-id="2"
                data-tag="allowRowEvents"
                id="cell-2-undefined"
                role="gridcell"
              >
                <div
                  data-tag="allowRowEvents"
                >
                  c4:d7:14:84:f8:cf
                </div>
              </div>
              <div
                class="sc-jrsJWt sc-kEqXSa sc-iqAclL eAcHFi dlLhQd eZwRKE rdt_TableCell"
                data-column-id="3"
                data-tag="allowRowEvents"
                id="cell-3-undefined"
                role="gridcell"
              >
                <div
                  data-tag="allowRowEvents"
                >
                  98962002
                </div>
              </div>
              <div
                class="sc-jrsJWt sc-kEqXSa sc-iqAclL eAcHFi dlLhQd eZwRKE rdt_TableCell"
                data-column-id="4"
                data-tag="allowRowEvents"
                id="cell-4-undefined"
                role="gridcell"
              >
                <div
                  data-tag="allowRowEvents"
                >
                  September 26th 2021, 3:10:43 pm
                </div>
              </div>
              <div
                class="sc-jrsJWt sc-kEqXSa sc-iqAclL eAcHFi spDng eZwRKE rdt_TableCell"
                data-column-id="5"
                data-tag="allowRowEvents"
                id="cell-5-undefined"
                role="gridcell"
              >
                <div
                  data-tag="allowRowEvents"
                >
                  <div
                    class="sc-jSFjdj iuSQRW"
                    data-testid="connection_type-container"
                    type="wifi"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      data-testid="icon-wifi"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M1 9l2 2c4.97-4.97 13.03-4.97 18 0l2-2C16.93 2.93 7.08 2.93 1 9zm8 8l3 3 3-3c-1.65-1.66-4.34-1.66-6 0zm-4-4l2 2c2.76-2.76 7.24-2.76 10 0l2-2C15.14 9.14 8.87 9.14 5 13z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div
                class="sc-jrsJWt sc-kEqXSa sc-iqAclL eAcHFi spDng eZwRKE rdt_TableCell"
                data-column-id="6"
                data-tag="allowRowEvents"
                id="cell-6-undefined"
                role="gridcell"
              >
                <div
                  data-tag="allowRowEvents"
                >
                  <div
                    class="sc-gKAaRy fwUWMS"
                    data-testid="status-container"
                    status="disconnected"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      data-testid="signal-off"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M21 1l-8.59 8.59L21 18.18V1zM4.77 4.5L3.5 5.77l6.36 6.36L1 21h17.73l2 2L22 21.73 4.77 4.5z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="sc-hBMUJo jzmWhL rdt_TableRow"
              id="row-2"
              role="row"
            >
              <div
                class="sc-jrsJWt sc-kEqXSa sc-iqAclL eAcHFi dlLhQd eZwRKE rdt_TableCell"
                data-column-id="1"
                data-tag="allowRowEvents"
                id="cell-1-undefined"
                role="gridcell"
              >
                <div
                  data-tag="allowRowEvents"
                >
                  <div
                    class="device_2"
                  >
                    device_2
                  </div>
                </div>
              </div>
              <div
                class="sc-jrsJWt sc-kEqXSa sc-iqAclL eAcHFi dlLhQd eZwRKE rdt_TableCell"
                data-column-id="2"
                data-tag="allowRowEvents"
                id="cell-2-undefined"
                role="gridcell"
              >
                <div
                  data-tag="allowRowEvents"
                >
                  a4:d7:14:84:f8:cf
                </div>
              </div>
              <div
                class="sc-jrsJWt sc-kEqXSa sc-iqAclL eAcHFi dlLhQd eZwRKE rdt_TableCell"
                data-column-id="3"
                data-tag="allowRowEvents"
                id="cell-3-undefined"
                role="gridcell"
              >
                <div
                  data-tag="allowRowEvents"
                >
                  98962003
                </div>
              </div>
              <div
                class="sc-jrsJWt sc-kEqXSa sc-iqAclL eAcHFi dlLhQd eZwRKE rdt_TableCell"
                data-column-id="4"
                data-tag="allowRowEvents"
                id="cell-4-undefined"
                role="gridcell"
              >
                <div
                  data-tag="allowRowEvents"
                >
                  September 26th 2021, 3:10:42 pm
                </div>
              </div>
              <div
                class="sc-jrsJWt sc-kEqXSa sc-iqAclL eAcHFi spDng eZwRKE rdt_TableCell"
                data-column-id="5"
                data-tag="allowRowEvents"
                id="cell-5-undefined"
                role="gridcell"
              >
                <div
                  data-tag="allowRowEvents"
                >
                  <div
                    class="sc-jSFjdj cBHKtd"
                    data-testid="connection_type-container"
                    type="cellular"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      data-testid="icon-cellular"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M17 4h3v16h-3zM5 14h3v6H5zm6-5h3v11h-3z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div
                class="sc-jrsJWt sc-kEqXSa sc-iqAclL eAcHFi spDng eZwRKE rdt_TableCell"
                data-column-id="6"
                data-tag="allowRowEvents"
                id="cell-6-undefined"
                role="gridcell"
              >
                <div
                  data-tag="allowRowEvents"
                >
                  <div
                    class="sc-gKAaRy cMtfuc"
                    data-testid="status-container"
                    status="connected"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      data-testid="signal-on"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M2 22h20V2z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
